<launch>
    <!-- Common Interface -->
    <arg name="device" default="192.168.0.102"/>
    <arg name="rate" default="20"/>

    <!-- Camera Settings -->
    <arg name="ip_address" default="$(arg device)"/>
    <arg name="camera_name" default="flir_ax5"/>
    <arg name="camera" default="thermal"/>
    <arg name="frame_id" default="$(arg camera)"/>
    <arg name="calib_url" default=""/>
    <arg name="fps" default="$(arg rate)"/>
    <arg name="raw" default="false"/>

    <!-- Node Settings -->
    <arg name="output" default="screen"/>
    <arg name="plugin" default="false"/>
    <arg name="proc" default="false"/>
    <arg name="view" default="false"/>

    <!-- Standalone -->
    <arg unless="$(arg plugin)"
        name="option" value="standalone flir_gige/FlirGigeNodelet"/>

    <!-- Manager -->
    <arg if="$(arg plugin)"
        name="manager" default="flir_gige_nodelet"/>
    <arg if="$(arg plugin)"
        name="option"
        value="load flir_gige/FlirGigeNodelet $(arg manager)"/>
    <node if="$(arg plugin)"
        pkg="nodelet" type="nodelet" name="$(arg manager)" args="manager"
        output="$(arg output)">
    </node>

    <!-- Nodelet -->
    <node pkg="nodelet" type="nodelet" name="$(arg camera)"
        output="$(arg output)" args="$(arg option)">
        <param name="identifier" type="string" value="$(arg ip_address)"/>
        <param name="camera_name" type="string" value="$(arg camera_name)"/>
        <param name="calib_url" type="string" value="$(arg calib_url)"/>
        <param name="frame_id" type="string" value="$(arg frame_id)"/>
        <param name="fps" type="double" value="$(arg fps)"/>
        <param name="raw" type="bool" value="$(arg raw)"/>
    </node>

    <!-- View -->
    <arg name="image" default="image_raw"/>
    <node if="$(arg view)"
        pkg="image_view" type="image_view" name="viewer"
        output="$(arg output)" ns="$(arg camera)" args="image:=$(arg image)">
    </node>
</launch>
